libexec/python/docker/api.py

```diff
@@ -225,7 +225,11 @@ def update_token(self, response=None, auth=None, expires_in=300):
 
         try:
             data = json.loads(response)
-            token = {"Authorization": "Bearer %s" % data["token"]}
+            if 'access_token' in data:
```

Comment:
OK, so I'm going to show some of my ignorance about this stuff.  Please bear with me.  

In the old version we always default to `data["token"]`.  In this PR, we default to `data["access_token"]` if it exists and only fall back to `data["token"]` in the event that it doesn't.  Cool.  

Now here is where my ignorance comes in.  Will there ever be a situation in which: 
1) `data["access_token"]` exists and 
2) it is *not* the correct token to use

In other words could both `access_token` and `token` exist in the same hash table in a situation where OAUTH2 should not really be used.  And then would the presence of `access_token` block the correct usage of `token`.

If this might be the case, maybe you should add a `try`.  If `access_token` fails fall back to `token` and try again.  

Is this just nutty?

Reply:
This code will be fine as if `access_token` is present you are using the OAUTH2 path, which doesn't set a `token`. `token` will appear alone if you use the docker token authentication path.

Reply:
Thanks for the clarification @dctrud !